### PR TITLE
[P2P] ValidationContext.BanDurationSeconds doesn't work as expected

### PR DESCRIPTION
**Pull request by quantumagi**
See work item 3059.

https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/3059

### CODE REVIEW

File: src/Stratis.Bitcoin/Connection/PeerBanning.cs [hunk 1]
```diff
@@ -71,6 +71,12 @@ public void BanAndDisconnectPeer(IPEndPoint endpoint, int banTimeSeconds, string
         {
             Guard.NotNull(endpoint, nameof(endpoint));
 
+            if (banTimeSeconds < 0)
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Connection/PeerBanning.cs]: This part I agree with

File: src/Stratis.Bitcoin/Connection/PeerBanning.cs [hunk 2]
```diff
@@ -98,7 +104,8 @@ public void BanAndDisconnectPeer(IPEndPoint endpoint, int banTimeSeconds, string
             foreach (var peerAddress in peerAddresses)
             {
                 peerAddress.BanTimeStamp = this.dateTimeProvider.GetUtcNow();
-                peerAddress.BanUntil = this.dateTimeProvider.GetUtcNow().AddSeconds(banTimeSeconds);
+                peerAddress.BanUntil = this.dateTimeProvider.GetUtcNow().AddSeconds(
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Connection/PeerBanning.cs]: If I remember correctly, we accept a zero second ban when we just want to disconnect a peer. I will try to find the issue where this last came up, but for now I'm not sure this logic is desirable.

Comment (quantumagi) [reply src/Stratis.Bitcoin/Connection/PeerBanning.cs]: Look at issue https://github.com/stratisproject/StratisBitcoinFullNode/issues/2815

File: src/Stratis.Bitcoin/Connection/PeerBanning.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin/Connection/PeerBanning.cs]: Ok

**Comment by dangershony** [comment]
I triggered tests again if they fail again @quantumagi please merge master where tests are stable now.

**Pull request merged**